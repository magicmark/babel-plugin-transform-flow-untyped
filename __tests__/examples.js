import fs from "mz/fs";
import { format } from "prettier";
import path from "path";
import { transformFileSync } from "babel-core";

it("examples pass tests", async () => {
  const exampleFileNames = fs
    .readdirSync(path.join(__dirname, "..", "./examples"))
    .filter(f => f.endsWith(".js"));

  for (let fileName of exampleFileNames) {
    const transform = transformFileSync(
      path.join(__dirname, "..", "examples", fileName),
      {
        parserOpts: {
          plugins: ["flow"]
        },
        babelrc: false,
        plugins: ["../lib/index"]
      }
    );

    const output = format(transform.code);
    expect(output).toMatchSnapshot();
  }
});
