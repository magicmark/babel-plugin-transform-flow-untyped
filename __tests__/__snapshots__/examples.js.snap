// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples pass tests 1`] = `
"// @flow

const helloWorld: string = \\"Helo World\\";
const numberOver: number = 9000;
const bears: boolean = false;
"
`;

exports[`examples pass tests 2`] = `
"// @flow

type Customer = {
  id: number,
  name: string
};

const getCustomer: (number, string, any) => Customer = (
  id: number,
  name: string,
  msg: any
): Customer => {
  console.log(msg);
  return { id, name };
};

const customer: Customer = getCustomer(1, \\"Alice\\", \\"Untyped Param\\");
"
`;
